fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/ContainmentIndexImpl.java

```diff
@@ -503,16 +521,17 @@ void setContainsLimit(final int limit) {
     @Override
     public String getContainedBy(@Nonnull final Transaction tx, final FedoraId resource) {
         final String resourceID = resource.getFullId();
-        final MapSqlParameterSource parameterSource = new MapSqlParameterSource();
-        parameterSource.addValue("child", resourceID);
-        final List<String> parentID;
+        final String parentID;
         if (tx.isOpenLongRunning()) {
-            parameterSource.addValue("transactionId", tx.getId());
-            parentID = jdbcTemplate.queryForList(PARENT_EXISTS_IN_TRANSACTION, parameterSource, String.class);
+            parentID = jdbcTemplate.queryForList(PARENT_EXISTS_IN_TRANSACTION, Map.of("child", resourceID,
+                    "transactionId", tx.getId()), String.class).stream().findFirst().orElse(null);
         } else {
-            parentID = jdbcTemplate.queryForList(PARENT_EXISTS, parameterSource, String.class);
+            parentID = this.getContainedByCache.get(resourceID, key ->
```

Comment:
You should be able to populate the cache as part of the `addContainedBy` call as well